universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -45,4 +60,47 @@ public Result edit(Request request, long programId, long blockDefinitionId) {
           String.format("Block ID %d not found for Program %d", blockDefinitionId, programId));
     }
   }
+
+  @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
+  public Result update(Request request, long programId, long blockDefinitionId) {
+    Form<BlockVisibilityPredicateForm> predicateFormWrapper =
+        formFactory.form(BlockVisibilityPredicateForm.class).bindFromRequest(request);
+
+    if (predicateFormWrapper.hasErrors()) {
+      StringBuilder errorMessageBuilder = new StringBuilder();
+      errorMessageBuilder.append("Did not save visibility condition:");
+      predicateFormWrapper
+          .errors()
+          .forEach(error -> errorMessageBuilder.append(String.format("\n• %s", error.message())));
+
+      return redirect(
+              routes.AdminProgramBlockPredicatesController.edit(programId, blockDefinitionId))
+          .flashing("error", errorMessageBuilder.toString());
+    } else {
+      BlockVisibilityPredicateForm predicateForm = predicateFormWrapper.get();
+      PredicateDefinition predicateDefinition =
```

Comment:
suggestion (blocking): Can we add a TODO here to implement more complex predicates?